### PR TITLE
Third home fix Vitalik Ischenko

### PR DESCRIPTION
**Pull request by futurestark**

### CODE REVIEW

**Comment by futurestark** [comment]
also  garrbage deleted

File: test/ExampleTests.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    EXPECT_EQ(true, result) << "Side Effect " << sideEffect << std::endl;
+
+    for (int i = 0; i < 16; ++i)
+     EXPECT_EQ(true, result) << "i = " << i << std::endl;
+    */
+
+TEST(ExampleTests, getUniqueWords){ 
+    
+
+	char filename[] = "test.txt";
```
Comment (smay1613) [inline test/ExampleTests.cpp:34]: std::string

File: test/ExampleTests.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    testStringWithResult = getUniqueWords(filename);
+	//std::cout << testStringWithResult << std::endl;
+    //EXPECT_THAT(resultVector, ::testing::UnorderedElementsAreArray(testVector)); 
+    //ASSERT_THAT(resultVector, ::testing::ElementsAre(5, 6, 7, 8, 9, 12, 13, 14, 15, 16));
+}
+
+TEST(ExampleTests, wordCounter){ 
+    
+
+	typedef std::multiset<std::string> mySet;
```
Comment (smay1613) [inline test/ExampleTests.cpp:49]: useless typedef, for aliases please use "using"

File: thirdHomeAdvanced.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+std::vector<std::string> getUniqueWords(char filename[])   //(const std::string& text);
+{ 
+    // Open a file stream 
+    std::fstream fs(filename); 
+  
+    // Create a map to store count of all words 
+    std::map<std::string, int> mp; 
+  
+    // Keep reading words while there are words to read
```
Comment (smay1613) [inline thirdHomeAdvanced.cpp:13]: comments are not needed, your code should be self-documented

File: thirdHomeAdvanced.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#include "thirdHomeAdvanced.hpp"
+
+
+std::vector<std::string> getUniqueWords(char filename[])   //(const std::string& text);
+{ 
+    // Open a file stream 
+    std::fstream fs(filename); 
+  
+    // Create a map to store count of all words 
+    std::map<std::string, int> mp;
```
Comment (smay1613) [inline thirdHomeAdvanced.cpp:11]: _naming_

File: thirdHomeAdvanced.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    // Create a map to store count of all words 
+    std::map<std::string, int> mp; 
+  
+    // Keep reading words while there are words to read 
+    std::string word; 
+    while (fs >> word) 
+    { 
+        // If this is first occurrence of word 
+        if (!mp.count(word)) 
+            mp.insert(make_pair(word, 1));
```
Comment (smay1613) [inline thirdHomeAdvanced.cpp:19]: _emplace(word, 1)_

**Pull request closed without merging**